src/components/form-field/text-input/text-input.tsx

```diff
@@ -15,7 +15,7 @@ const formFieldGroupStyles = classNames(
     `[.group.form-field-group_&:last-child_.target-field]:border-l-1 [.group.form-field-group_&:last-child_.target-field]:rounded-l-none`
 );
 
-export interface TextInputProps {
+export interface TextInputProps extends HTMLProps<HTMLInputElement> {
```

Comment:
Have we already done this kind of thing in other places?

I'm not sure if we want to expose all the underlying html props and rather keep an abstraction later?

Otherwise logic like `type={isSubmitButton ? "submit" : "button"}` doesn't make much sense since you should just set the native `type` element.

Reply:
You're right. We're not accepting the rest of the input props, so extending from `HTMLProps<HTMLInputElement>` doesn't make sense. I made the change 👌

Reply:
I'm a bit against this kind of abstraction. To me, the component should follow the underlying HTML interface so it's more compatible with other libraries that assume that an `onBlur` function follows the HTML input signature. Take, for example, react-hook-form. You could eventually do something like this `(field) => <TextInput {...field} />` (which is the same as  (`<TextInput onChange={field.onChange} onBlur={filed.onBlur} value={field.value} />`) but for that to work, the TextInput component should follow the HtmlInput interface. In addition, I haven't seen so far a components library that abstracts or modifies the html element's props, which makes me think that it's not the way to go if we want this to be used one day by more people 🤔

Reply:
Completely agree with all those arguments 😄 

The counter-argument is when we come up with a new type of input which doesn't have an html equivalent. For a contrived example a `<RetinaInput />` which still fires `onBlur`, `onChange` etc but obviously doesn't have a native event type we can use. In this case we have to come up with a custom event anyway which will then be inconsistent with our existing pattern.

But I think your arguments for keeping the native events are stronger 😄

Reply:
Another reason to categorize components into `atoms` and `molecules`, so that atoms can be closer to the native events 😬